salt/states/aptpkg.py

```diff
@@ -14,11 +14,21 @@
 log = logging.getLogger(__name__)
 
 
+# Define the module's virtual name
+__virtualname__ = 'apt'
+
+
 def __virtual__():
     '''
     Only work on apt-based platforms with pkg.get_selections
     '''
-    return 'pkg.get_selections' in __salt__
+    try:
+        __salt__['pkg.get_selections']
+    except KeyError:
+        # direct access not to iterate over lazyloader
+        return False
+    else:
+        return __virtualname__
 
```

Comment:
The problem with this is checking for membership in `__salt__` causes the LazyLoader to load everything.